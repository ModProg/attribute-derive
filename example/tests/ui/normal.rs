use example::Normal;

#[derive(Normal)]
struct None;

#[derive(Normal)]
#[ident(example = 2.)]
struct Example;

#[derive(Normal)]
#[ident(
    example = 2.,
    optional_implicit = 10,
    optional_default = 3,
    default = 2,
    conflict_a = "hello",
    mandatory_flag
)]
struct ExampleOI;

#[derive(Normal)]
#[ident(example = 1., mandatory_flag)]
#[a(conflict_a = "hey")]
#[b(conflict_b = "hi")]
struct Conflict;

#[derive(Normal)]
#[ident(hello)]
#[single(hello)]
#[empty(hello)]
struct UnknownField;

fn main() {}
