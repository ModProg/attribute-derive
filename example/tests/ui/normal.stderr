error: required `example` is not specified

       = help: try `#[ident(example = 2.5)]`
 --> tests/ui/normal.rs:3:10
  |
3 | #[derive(Normal)]
  |          ^^^^^^
  |
  = note: this error originates in the derive macro `Normal` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `conflict_a` conflicts with mutually exclusive `conflict_b`
  --> tests/ui/normal.rs:22:5
   |
22 | #[a(conflict_a = "hey")]
   |     ^^^^^^^^^^

error: `conflict_b` conflicts with mutually exclusive `conflict_a`
  --> tests/ui/normal.rs:23:5
   |
23 | #[b(conflict_b = "hi")]
   |     ^^^^^^^^^^

error: supported fields are `optional_implicit`, `optional_explicit`, `optional_default`, `default`, `conflict_a`, `conflict_b`, `example` and `flag`
  --> tests/ui/normal.rs:27:9
   |
27 | #[ident(hello)]
   |         ^^^^^

error: expected empty attribute
  --> tests/ui/normal.rs:29:9
   |
29 | #[empty(hello)]
   |         ^^^^^

error: expected supported field `field`
  --> tests/ui/normal.rs:28:10
   |
28 | #[single(hello)]
   |          ^^^^^
